src/utils/domUtils.js

```diff
@@ -138,6 +138,25 @@ function getPosition(elem, offsetParent) {
 }
 
 /**
+ * Get an element's size
+ *
+ * @param {HTMLElement} elem
+ * @returns {{width: number, height: number}}
+ */
+function getSize(elem) {
+  let rect = {
+    width: elem.offsetWidth,
+    height: elem.offsetHeight
+  };
+  if (typeof elem.getBoundingClientRect !== 'undefined') {
+    let {width, height} = elem.getBoundingClientRect();
+    rect.width = width || rect.width;
```

Comment:
Right, it's the case where `offsetWidth` is `null`-y rather than 0 that I'm concerned about here.